### PR TITLE
Reads happen inside a transaction

### PR DESCRIPTION
**Pull request by sebastianlavender**

### CODE REVIEW

File: pipe-storage-postgresql/src/main/java/com/tesco/aqueduct/pipe/storage/PostgresqlStorage.java [hunk 1]
```diff
@@ -464,7 +465,7 @@ private static String getCompactionQuery() {
     }
 
     private static String getCompactDeletionQuery() {
-        return "DELETE FROM events WHERE created_utc <= ? AND data IS NULL;";
+        return "DELETE FROM events WHERE created_utc <= ? AND data IS NULL LIMIT ;";
```
Comment (adam-yorwerth) [inline pipe-storage-postgresql/src/main/java/com/tesco/aqueduct/pipe/storage/PostgresqlStorage.java:468]: Don't think this is meant to be here

File: pipe-storage-postgresql/src/main/java/com/tesco/aqueduct/pipe/storage/PostgresqlStorage.java [hunk 2]
```diff
@@ -131,6 +131,8 @@ private String getSelectLocationGroupsQuery() {
     private Connection getConnection() throws SQLException {
         long start = System.currentTimeMillis();
         Connection connection = pipeDataSource.getConnection();
+        connection.setTransactionIsolation(Connection.TRANSACTION_REPEATABLE_READ);
+        connection.setAutoCommit(false);
```
Comment (adam-yorwerth) [inline pipe-storage-postgresql/src/main/java/com/tesco/aqueduct/pipe/storage/PostgresqlStorage.java:135]: If we're including this in the method, can we rename it to `getConnectionAndStartTransaction` or something?

File: pipe-storage-postgresql/src/main/java/com/tesco/aqueduct/pipe/storage/PostgresqlStorage.java (hunk 1 above)
Comment (sebastianlavender) [reply pipe-storage-postgresql/src/main/java/com/tesco/aqueduct/pipe/storage/PostgresqlStorage.java:468]: It is not and had just removed it from branch

**Review by LorenzoCianciaruso — DISMISSED** [review]
LGTM

**Review by adam-yorwerth — APPROVED** [review]

**Review by adam-yorwerth — APPROVED** [review]

**Pull request merged**